google-cloud-debugger/lib/google/cloud/debugger/breakpoint/evaluator.rb

```diff
@@ -909,6 +917,21 @@ def wrap_expression expression
             end
 
             ##
+            # @private Evaluation tracing callback function for line event.
+            # This is called when ever line of Ruby code invoked during tracing.
+            # It keeps track of expression evaluation time usage and error out
+            # if the evaluation exceeds the predetermined deadline.
+            def trace_line_callback
+              deadline = Thread.current.thread_variable_get :evaluation_deadline
+
+              return if Time.now <= deadline
```

Comment:
Is there a way to test the expression evaluation threshold? I don't see this case in the tests.